app/src/main/java/com/iic/shopingo/ui/HomeActivity.java

```diff
@@ -41,27 +101,24 @@ private void init() {
 
   @Override
   public boolean onCreateOptionsMenu(Menu menu) {
-    // Inflate the menu; this adds items to the action bar if it is present.
     getMenuInflater().inflate(R.menu.menu_home, menu);
     return true;
   }
 
   @Override
   public boolean onOptionsItemSelected(MenuItem item) {
-    // Handle action bar item clicks here. The action bar will
-    // automatically handle clicks on the Home/Up button, so long
-    // as you specify a parent activity in AndroidManifest.xml.
     int id = item.getItemId();
 
-    //noinspection SimplifiableIfStatement
-    if (id == R.id.action_settings) {
-      return true;
-    } else if (id == R.id.action_logout) {
-      SharedUserConnector.getInstance().logout(this);
-      navigateToOnboarding();
-    } else if (id == R.id.action_update_contact) {
-      Intent intent = new Intent(this, ContactDetailsActivity.class);
-      startActivity(intent);
+    switch (id) {
+      case R.id.action_logout:
+        FacebookConnector.logout(this);
+        CurrentUser.getInstance().logout();
```

Comment:
As above, we need a more robust state handler. Until then there's no point in doing half solutions.